### PR TITLE
feat(codepipeline): allow granting manual approval permissions

### PR DESCRIPTION
**Pull request by hoegertn**
This feature allows granting permissions to click on the manual approval button in a CodePipeline to a supplied IGrantable. This can be used to allow IAM users and roles to approve pipeline actions.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-codepipeline-actions/README.md [hunk 1]
```diff
@@ -866,6 +866,8 @@ but `notifyEmails` were,
 a new SNS Topic will be created
 (and accessible through the `notificationTopic` property of the Action).
 
+If you want to grant any principal permissions to click on the approve button you can invoke the method `grantManualApproval` with the given `IGrantable`.
+
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/README.md:870]: Some example code here would be nice 🙂.

File: packages/@aws-cdk/aws-codepipeline-actions/lib/manual-approval-action.ts [hunk 2]
```diff
@@ -46,6 +47,7 @@ export class ManualApprovalAction extends Action {
    * a new Topic will be created.
    */
   private _notificationTopic?: sns.ITopic;
+  private stage?: codepipeline.IStage;
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/lib/manual-approval-action.ts:50]: Let's move this below `props`.

File: packages/@aws-cdk/aws-codepipeline-actions/lib/manual-approval-action.ts [hunk 3]
```diff
@@ -63,8 +65,32 @@ export class ManualApprovalAction extends Action {
     return this._notificationTopic;
   }
 
-  protected bound(scope: Construct, _stage: codepipeline.IStage, options: codepipeline.ActionBindOptions):
-  codepipeline.ActionConfig {
+  /**
+   * grantManualApproval
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/lib/manual-approval-action.ts:69]: Some docs here would be nice 🙂.

File: packages/@aws-cdk/aws-codepipeline-actions/lib/manual-approval-action.ts [hunk 4]
```diff
@@ -63,8 +65,32 @@ export class ManualApprovalAction extends Action {
...
+   *
+   * https://docs.aws.amazon.com/codepipeline/latest/userguide/approvals-iam-permissions.html
+   *
+   * @param grantable the grantable to attach the permissions to
+   */
+  public grantManualApproval(grantable: iam.IGrantable) {
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/lib/manual-approval-action.ts:75]: Please specify the return type explicitly.

File: packages/@aws-cdk/aws-codepipeline-actions/lib/manual-approval-action.ts [hunk 5]
```diff
@@ -63,8 +65,32 @@ export class ManualApprovalAction extends Action {
...
+    if (!this.stage || !this.stage.pipeline) {
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/lib/manual-approval-action.ts:76]: `stage.pipeline` is never `undefined`, so this condition can be simplified to:

```suggestion
    if (!this.stage) {
```

File: packages/@aws-cdk/aws-codepipeline-actions/test/manual-approval.test.ts [hunk 6]
```diff
@@ -25,6 +26,123 @@ nodeunitShim({
       test.done();
     },
 
+    'allows granting manual approval permissions to role'(test: Test) {
+      const stack = new Stack();
+      const role = new iam.Role(stack, 'Human', { assumedBy: new iam.AnyPrincipal() });
+      const manualApprovalAction = new cpactions.ManualApprovalAction({
+        actionName: 'Approve',
+      });
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/test/manual-approval.test.ts:34]: Can you move these local variables closer to where they are first used?

File: packages/@aws-cdk/aws-codepipeline-actions/test/manual-approval.test.ts [hunk 7]
```diff
@@ -25,6 +26,123 @@ nodeunitShim({
...
+                    ':',
+                    {
+                      'Ref': 'AWS::AccountId',
+                    },
+                    ':',
+                    {
+                      'Ref': 'pipelineDBECAE49',
+                    },
+                  ],
+                ],
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/test/manual-approval.test.ts:87]: Let's conserve the lines here a little bit more:

```suggestion
                'Fn::Join': ['', [
                  'arn:',
                  {'Ref': 'AWS::Partition' },
                  ':codepipeline:',
                  {'Ref': 'AWS::Region' },
                  ':',
                  {'Ref': 'AWS::AccountId' },
                  ':',
                  {'Ref': 'pipelineDBECAE49'},
                ]],
```

Same comment below.

**Review by skinny85 — CHANGES_REQUESTED** [review]
Looks great @hoegertn, thanks a lot for the contribution!

I wouldn't be the picky asshole I'm known for being if I didn't find a few places to nitpick on 😉.

**Comment by hoegertn** [comment]
@skinny85 I addressed your comments. Something went sideways with the push but the changes seem to be correct. Squashing should make it go away.

**Review by skinny85 — APPROVED** [review]
👍

**Pull request merged**